doc/generate-kuadrant-auth-policy.md

```diff
@@ -73,6 +78,91 @@ spec:
                 method: GET
 ```
 
+### `apiKey` Type Description
+
+The following OAS example has one protected endpoint `GET /dog` with `apiKey` security scheme type.
+
+```yaml
+paths:
+  /dog:
+    get:
+      operationId: "getDog"
+      security:
+        - securedDog: []
+      responses:
+        405:
+          description: "invalid input"
+components:
+  securitySchemes:
+    securedDog:
+      type: apiKey
+      name: dog_token
+      in: query
+```
+
+Running the command
+
+```
+kuadrantctl generate kuadrant authpolicy --oas ./petstore-openapi.yaml  | yq -P
+```
+
+The generated authpolicy (only relevan fields shown here):
+
+```yaml
+kind: AuthPolicy
+apiVersion: kuadrant.io/v1beta2
+metadata:
+  name: petstore
+  namespace: petstore
+  creationTimestamp: null
+spec:
+  routeSelectors:
+    - matches:
+        - path:
+            type: Exact
+            value: /dog
+          method: GET
+  rules:
+    authentication:
+      getDog_securedDog:
+        credentials:
+          queryString:
+            name: dog_token
+          apiKey:
+            selector:
+              matchLabels:
+                kuadrant.io/apikeys-by: securedDog
+        routeSelectors:
+          - matches:
+              - path:
+                  type: Exact
+                  value: /dog
+                method: GET
+```
+
+In this particular example, the endpoint `GET /dog` will be protected.
+The token needs to be in the query string of the request included in a parameter named `dog_token`.
+Kuadrant will validate received tokens against tokens found in kubernetes secrets with label `kuadrant.io/apikeys-by: ${sec scheme name}`.
+In this particular example the label selector will be: `kuadrant.io/apikeys-by: securedDog`.
+
+Like the following example:
+
+```yaml
```

Comment:
one for later maybe, but do you think it'd be nice to generate an example secret too? not a big deal, just something I thought of while using

Reply:
you mean that the command `kuadrantctl generate kuadrant authpolicy` generates the authpolicy and additionally some secret? That could be done. Or write in stderr (stdout will likely be used piped with kubectl) a example secret too.

Reply:
This sounds good to you?


```
❯ bin/kuadrantctl generate kuadrant authpolicy --oas examples/oas3/petstore-multiple-sec-requirements.yaml 1>/dev/null
======================================================================================================
POST /v1/cat endpoint is protected with ApiKey. Consider creating secrets with valid tokens
---
apiVersion: v1
kind: Secret
metadata:
  creationTimestamp: null
  labels:
    authorino.kuadrant.io/managed-by: authorino
    kuadrant.io/apikeys-by: api_key
  name: api_key
stringData:
  api_key: MY_SECRET_TOKEN_VALUE
type: Opaque

======================================================================================================
GET /v1/snake endpoint is protected with ApiKey. Consider creating secrets with valid tokens
---
apiVersion: v1
kind: Secret
metadata:
  creationTimestamp: null
  labels:
    authorino.kuadrant.io/managed-by: authorino
    kuadrant.io/apikeys-by: snake_token
  name: snake_token
stringData:
  api_key: MY_SECRET_TOKEN_VALUE
type: Opaque
```
> Note: `stdout` has been redirected to `/dev/null`

Reply:
https://github.com/Kuadrant/kuadrantctl/pull/51